plugin/tools.py

```diff
@@ -175,31 +195,56 @@ def folder(self):
         return path.dirname(self.__full_path)
 
     def loaded(self):
-        """Is the file loaded?
-        """
+        """Check if the file is loaded."""
         if self.__full_path:
             return True
         return False
 
+    @staticmethod
+    def is_unchanged(file_path):
+        if not file_path:
+            return False
+        actual_mod_time = path.getmtime(file_path)
+        if file_path not in File.__modification_cache:
+            log.debug(" never seen file '%s' before. Updating.", file_path)
+            File.__modification_cache[file_path] = actual_mod_time
+            return False
+        cached_mod_time = File.__modification_cache[file_path]
+        if actual_mod_time > cached_mod_time:
```

Comment:
Probably better to check using `!=`. Modification dates can change both ways. For example when checking out older branch.